### PR TITLE
More docs on NBTs 

### LINKED ISSUE #855: Add content about the little endian network byte order for nbts
NBTs use the little endian network byte order in the bedrock protocol. This is a very important part of NBTs that isn't mentioned at all in the documentation.

It would be very nice to add this.

### PR DESCRIPTION
**Pull request by theaddonn**
- Added Documentation on Network-Little-Endian integer encoding (https://github.com/Bedrock-OSS/bedrock-wiki/issues/855)
- Fixed several typos
- Cleaned up layout
- Added more precise descriptions for Datatypes
- Added list of existing NBT libraries sorted by programming language

Further questions:
- Should I also document extra encodings not commonly found in Bedrock Edition such as `Big Endian` and `Network Big Endian`?
- Is there any way to enable sorting for tables?

### CODE REVIEW

**Review by SmokeyStack — CHANGES_REQUESTED** [review]
Really not a fan of the random new lines in the markdown files. While it won't affect rendering, it is vastly formatted differently than our other files.

**Comment by theaddonn** [comment]
Are you sure? I just did it because it made editing easier.. long lines are hard to read in your editor and can be difficult in the long run (This auto formatting was even directly done by my ide, sorrry). But I can surely change it

**Comment by theaddonn** [comment]
Should be all fixed now, pretty as we like it

File: docs/nbt/nbt-in-depth.md [hunk 1]
```diff
@@ -5,85 +5,119 @@ mentions:
...
+### Reading Compounds
+Compound has all properties named, so when reading a property, it is always necessary to read its name as well. The procedure for reading Compound is rather simple.
+First, we read the type, the type can be anything, but if it is equal to an empty byte, then it is the end of the compound and then we jut stop reading, but if the type is not equal to the Compound Ending tag,
+then the significant type of the property that we will read. The read property is always followed by the name (key), which needs to be read as a [String](#reading-Strings), and after the String is read, then we can read value.
```
Comment (SmokeyStack) [inline docs/nbt/nbt-in-depth.md]: `reading-Strings` -> `reading-strings`

File: docs/nbt/nbt-in-depth.md [hunk 2]
```diff
@@ -5,85 +5,119 @@ mentions:
...
 
 ## Minecraft BE NBT files
```
Comment (SmokeyStack) [inline docs/nbt/nbt-in-depth.md]: Changed `Minecraft BE` to `Bedrock`

File: docs/nbt/nbt-in-depth.md [hunk 3]
```diff
@@ -5,85 +5,119 @@ mentions:
...
-When reading Minecraft NBT files, it is always important to be careful if there is no Bedrock Header at the beginning of the file, see [Bedrock NBT Header](#bedrock-nbt-file-header), but not all MCBE NBT files contain this header, for example `.mcstructure` also does not contain a Bedrock NBT header, unlike `level.dat`.
-You also need to pay attention to the root element in the file, i.e. the list or compoud,
-The root element also looks like a property, so you need to read the name of this root property, although Bedrock does not use these names, so these names are empty, but they are there.
-Here is how .mcstructure looks like where JSON represents NBT.
+When reading Minecraft NBT files, it is always important to be careful if there is no Bedrock Header at the beginning of the file, see [Bedrock NBT Header](#bedrock-nbt-file-header), but not all MCBE NBT files contain this header,
```
Comment (SmokeyStack) [inline docs/nbt/nbt-in-depth.md]: Likewise, change all `MCBE` to just `Bedrock`

File: docs/nbt/nbt-in-depth.md [hunk 4]
```diff
@@ -5,85 +5,119 @@ mentions:
...
+That's why we recommend first understanding NBT and learning to read it correctly, then it won't be difficult to write NBT.
 
 ## Bedrock NBT File header
-The NBT bedrock Header is indicated by two 4-byte numbers, the first is always 8 and the second indicates the size of the nbt structure in bytes. E.g.
+The NBT bedrock Header is indicated by two 4-byte numbers, the first is always 8, and the second indicates the size of the nbt structure in bytes.
+E.g.
     - `08 00 00 00` - `bf 00 00 00`
     - < always 8 > < always the size of the NBT structure - exclude headers 8 bytes>
 
 ## Little-Endian
```
Comment (SmokeyStack) [inline docs/nbt/nbt-in-depth.md]: Does little endian need a `-`?

File: docs/nbt/nbt-in-depth.md [hunk 5]
```diff
@@ -5,85 +5,119 @@ mentions:
...
+It may seem illogical, but little-endian is almost always used when writing and reading from files. A single `byte` is the same in both methods because it is one byte in size.
+For example:
 - Int64 (long) `0x11223344aabbccdd`
 - Split to 8 bytes `0x11 0x22 0x33 0x44 0xAA 0xBB 0xCC 0xDD`
 - Reverse `0xDD 0xCC 0xBB 0xAA 0x44 0x33 0x22 0x11`
 - Write `dd cc bb aa   44 33 22 11`
-- Done (when reading the number just go backwards this example.)
+- Done (when reading the number just goes backwards this example.)
+
+## Network-Little-Endian
```
Comment (SmokeyStack) [inline docs/nbt/nbt-in-depth.md]: Does network little endian need a `-`?

**Review by SmokeyStack — CHANGES_REQUESTED** [review]

File: docs/nbt/nbt-in-depth.md (hunk 4 above)
Comment (theaddonn) [reply docs/nbt/nbt-in-depth.md]: nope, and it is removed now

**Comment by theaddonn** [comment]
All resolved now, but still..
Should I also document extra encodings not commonly (aka Java only) found in Bedrock Edition such as `Big Endian` and `Network Big Endian`?

**Review by SmokeyStack — APPROVED** [review]

**Comment by SmokeyStack** [comment]
> All resolved now, but still.. Should I also document extra encodings not commonly (aka Java only) found in Bedrock Edition such as `Big Endian` and `Network Big Endian`?

We can do this in a different PR. Going to merge this PR now.

**Pull request merged**